lib/util/auth.js

```diff
@@ -0,0 +1,105 @@
+import { push } from 'connected-react-router'
+
+/**
+ * Custom links under the user account dropdown.
+ * TODO: Determine/implement the urls below.
+ */
+export const accountLinks = [
+  {
+    text: 'My Account',
+    url: 'account'
+  },
+  {
+    text: 'Help',
+    url: 'help'
+  }
+]
+
+/**
+ * Handle logging out with Auth0 and return to the provided path.
+ */
+export function logout (path = '') {
+  const returnTo = encodeURIComponent(`${getAuthRedirectUri()}${path}`)
+  window.location.href = `https://${process.env.AUTH0_DOMAIN}/v2/logout?returnTo=${returnTo}&client_id=${process.env.AUTH0_CLIENT_ID}`
+}
+
+export const getAuthRedirectUri = () => {
+  if (typeof window !== 'undefined') {
+    const url = `${window.location.protocol}//${window.location.host}`
+    console.log(url)
+    return url
+  }
+  return 'http://localhost:9966'
+}
+
+/**
+ * Obtains the Auth0 {domain, audience, clientId} configuration, if the following applies in config.yml:
+ * - persistence is defined,
+ * - persistence.enabled is true,
+ * - persistence.strategy is defined and is not 'localStorage',
+ * - persistence.auth0 is defined.
+ * @param otpConfig The OTP configuration params from config.yml.
+ * @returns The Auth0 configuration, or null if the conditions are not met.
+ */
+export function getAuth0Config (otpConfig) {
+  const { persistence } = otpConfig
+  if (persistence) {
+    const { enabled = false, strategy = null, auth0 = null } = persistence
+    return (enabled && strategy && strategy !== 'localStorage') ? auth0 : null
```

Comment:
This all seems very specific to Auth0, so I think we should make the strategy required to be "Auth0" here.

Reply:
I don't think the strategy itself should be set to Auth0. In the configuration, that is set to `otp_middleware` (you might have middleware with different authentication methods). The auth0-specific check is for `persistence.auth0` (in the deconstructed var declarations from `presistence`).
```yml
persistence:
  enabled: true
  strategy: otp_middleware
  auth0:
    audience: https://otp-middleware
    domain:
    clientId:
```

Reply:
I like that you've clarified the way the config should be setup. However, in the code there is still a check that the strategy isn't `localStorage`. I'd prefer this to directly check if the strategy is `otp-middleware`.

Reply:
Yes, I agree. Checking that something is not `localhost` is just asking for bugs down the road.